Features/add prerequisites checkboxes #162769912

software-project opened the pull request:
https://www.pivotaltracker.com/n/projects/1405352/stories/162769912

This one will be ready to test when salesforce part is deployed.

akegan commented on app/assets/json/translations/locale-en.json:
```diff
@@ -442,6 +442,11 @@
         "APPLICANT_SEXUAL_ORIENTATION": "How do you describe your sexual orientation or sexual identity?",
         "APPLICATION_CONTACT_DETAILS": "Application Contact Details",
+        "APPLICATION_IS_FIRST_TIME_HOMEBUYER": "I have not owned property within the past three years.",
+        "APPLICATION_HAS_COMPLETED_HOMEBUYER_EDUCATION": "I have completed homebuyers’ education.",
```
minor edit on this. We just got feedback from MOHCD that it's actually *homebuyer education* rather than *homebuyers' education*

akegan commented on app/assets/json/translations/locale-en.json:
```diff
@@ -442,6 +442,11 @@
+        "APPLICATION_HAS_COMPLETED_HOMEBUYER_EDUCATION": "I have completed homebuyers’ education.",
+        "APPLICATION_VISIT_HOMEOWNERSHIP_CALENDAR": "If you haven't done this yet, visit the <a href='{{url}}' target='_blank'>HomeownershipSF calendar</a> to attend a workshop near you.",
+        "APPLICATION_HAS_LOAN_PRE_APPROVAL": "I am pre-approved for a loan by a MOHCD-approved lender.",
+        "APPLICATION_VISIT_MOHCD": "If you haven't done this yet, visit the <a target='{{url}}'>MOHCD website</a> to find a list of approved lenders",
```
This link will be off-the application too so we should open it in a new tab with target='_blank'

akegan commented on spec/e2e/features/short_form/filling_sale_application.feature:
```diff
@@ -7,3 +7,5 @@ Feature: Short Form Application
         Given I go to the first page of the "Sale Test Listing" application
         When I hit the Next button "1" time
         Then I should be on prerequisites page
+        And I check complete homebuyers education
```
All 3 of the checkboxes should be required on this page.

software-project commented:
@akegan I moved requiring checkboxes from the other ticket here and addressed your minor issues. Regarding the 6 min place issue I think I saw it before a while ago on a test listing, so I assumed it's a data issue.

akegan commented:
@software-project I think that we can't merge this PR until we are able to complete an entire application for an ownership listing. I'll go ahead and pull the bug ticket into this sprint https://www.pivotaltracker.com/story/show/164322855

akegan commented:
@software-project let me know when you've pushed up the fix with stubbing householdMatch: true and I'll QA again

akegan commented on app/assets/javascripts/short-form/ShortFormApplicationController.js.coffee:
```diff
@@ -551,6 +551,9 @@ ShortFormApplicationController = (
       return
     ShortFormApplicationService.checkHouseholdEligiblity($scope.listing)
       .then( (response) ->
+        # TODO remove this then SF household API is fixed for sale listing
```
```suggestion
        # TODO remove this when SF household API is fixed for sale listing
```
